Vagrantfile

```diff
@@ -12,20 +12,63 @@ end
 # Vagrantfile API/syntax version. Don't touch unless you know what you're doing!
 VAGRANTFILE_API_VERSION = "2"
 
+def get_amis
+  require "open-uri"
+  require "json"
+
+  images_uri = "https://dl.flynn.io/ec2/images.json"
+  json = JSON.parse(open(images_uri).read())
+  latest = json["versions"][0]["images"]
+
+  Hash[ latest.map {|ami| [ ami["region"], ami["id"] ] } ]
+end
+
 Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   config.vm.box = "flynn-base"
   config.vm.box_url = "https://dl.flynn.io/vagrant/flynn-base.json"
   config.vm.box_version = "> 0"
 
-  # RFC 5737 TEST-NET-1 used to avoid DNS rebind protection
-  config.vm.network "private_network", ip: "192.0.2.100"
-
-  config.vm.provider "virtualbox" do |v|
+  # VAGRANT_MEMORY          - instance memory, in MB
+  # VAGRANT_CPUS            - instance virtual CPUs
+  config.vm.provider "virtualbox" do |v, override|
     v.memory = ENV["VAGRANT_MEMORY"] || 1024
     v.cpus = ENV["VAGRANT_CPUS"] || 2
+
+    # RFC 5737 TEST-NET-1 used to avoid DNS rebind protection
+    override.vm.network "private_network", ip: "192.0.2.100"
+  end
+
+  # AWS_ACCESS_KEY_ID       - AWS IAM public ID
+  # AWS_SECRET_ACCESS_KEY   - AWS IAM secret key
+  # AWS_SESSION_TOKEN       - AWS IAM role session token
+  # AWS_DEFAULT_REGION      - AWS EC2 region to launch instance in
+  # FLYNN_VAGRANT_KEY       - SSH key (.pem file) used to connect to the instance
+  # FLYNN_VAGRANT_SUBNET    - AWS VPC subnet to launch instance in
+  # FLYNN_VAGRANT_IAM       - AWS IAM role to give instance permissions
+  # FLYNN_VAGRANT_TAGS      - AWS EC2 tags to apply to instance (comma separated key-value pairs)
+  config.vm.provider "aws" do |aws, override|
+    aws.access_key_id = ENV["AWS_ACCESS_KEY_ID"]
+    aws.secret_access_key = ENV["AWS_SECRET_ACCESS_KEY"]
+    aws.session_token = ENV["AWS_SESSION_TOKEN"]
+    aws.keypair_name = File.basename(ENV["FLYNN_VAGRANT_KEY"]).split('.')[0] rescue nil
+
+    get_amis().each_pair { |ami,region| aws.region_config region, :ami => ami }
+
+    aws.region = ( ENV["AWS_DEFAULT_REGION"] || "us-east-1" )
+
+    aws.instance_type = "m3.large"
+    aws.subnet_id = ENV["FLYNN_VAGRANT_SUBNET"]
+    aws.iam_instance_profile_arn = ENV["FLYNN_VAGRANT_IAM"]
+    added_tags = Hash[ ENV["FLYNN_VAGRANT_TAGS"].split(",").map {|kv| kv.split("=")} ] rescue {}
```

Comment:
`Hash#[]` always returns `{}`. It will of course fail if there's an odd number of arguments, but I'd argue returning an error there is more intuitive than hiding the failure.